### ISSUE TITLE
Multi-Targets configuration and abstract_target build

### ISSUE THREAD

**Issue by lifely** [CocoaPods/CocoaPods#4863]
Hello,

I had to update my podfile today, following 1.0 update and changes.
I have a project with multiple target that represents my environements.

I previously didn't specified any target in the podfile like so : 
```
# Specs Sources List
source 'https://github.com/CocoaPods/Specs.git'

# Project Configuration
xcodeproj 'project.xcodeproj'
platform :ios, '8.0'

## Settings
inhibit_all_warnings!
use_frameworks!

pod ...
pod ...
```

And i figured out that this configuration doesn't work for multiple targets anymore.
So i looked around and found that the easiest way to define the same pods for all the targets was to use an `abstract_target` like so : 

```
abstract_target 'defaults' do

  ## - Networks
  pod 'Alamofire', '~> 2.0.2'
  pod 'ReachabilitySwift', '~> 2.3.3'

target 'Target'
target 'Target2'
...
```

However by doing so i saw my compile time explode, this solution made cocoapods generate an individual library for each targets.

![image](https://cloud.githubusercontent.com/assets/442276/12824643/8689a50c-cb72-11e5-9e68-6543f0cd85ef.png)

I wonder what is possible now, to configure cocoapods to have a set of dependencies but only build one framework used for every targets.
The way i see the abstract_target feature would be to create a Pods framework to that name and link it against all dependents target.

Am i missing something ?

**Comment by segiddins** [comment]
Are you sure all of those targets have the same platform?

**Comment by lifely** [comment]
Hey @segiddins thx for the reply.

Well, i'm pretty sure i haven't done anything facy with the platforms, all of my app targets are on `standard` architectures and `iOS` supported platforms.

Also, my podfile example aren't changed much compared to my own podfile, the only thing i've done here is anonymize the project.

Thanks for the time.

**Comment by segiddins** [comment]
@mrackwitz want to look into this since its deduplication related?

**Comment by mrackwitz** [comment]
Is deduplication in your `~/.cocoapods/config.yaml` disabled?
Do you have anywhere in your Podfile a further `target` block, which defines additional dependencies, which intersect with transitive dependencies of those defined in your `abstract_target` block?

**Comment by lifely** [comment]
Here my podfile : 

```
# Specs Sources List
source 'https://github.com/CocoaPods/Specs.git'

# Project Configuration
xcodeproj 'Project.xcodeproj'
platform :ios, '8.0'

## Settings
inhibit_all_warnings!
use_frameworks!

## Remember
## If you aren't using frameworks, after adding a Pod here,
## add it to the bridging header to expose it to Swift

abstract_target 'defaults' do

  ## - Networks
  pod 'Alamofire', '~> 2.0.2'
  pod 'ReachabilitySwift', '~> 2.3.3'

  ## - Parsing, Encoding, Decoding
  pod 'SwiftyJSON', '~> 2.3.2'
  pod 'CryptoSwift', '~> 0.2.2'

  ## - Layout
  pod 'Cartography', '~> 0.6.0'

  ## - Algorithm
  pod 'Dollar', '~> 4.1.0'

  ## - Librairies
  pod 'FontAwesomeKit/Core', '~> 2.2.0'

  pod 'Crashlytics', '~> 3.6.0'
  pod 'Fabric', '~> 1.6.4'

  ## - UI Components
  pod 'JDStatusBarNotification', '~> 1.5.3'
  pod 'Kingfisher', '~> 2.0.1'
  pod 'Armchair', '~> 0.1.1'

  target 'Project'
  target 'Project Recette'
  target 'Project Recette 2'
  target 'Project Recette 3'
  target 'Project Recette 8'
  target 'Project Pre-Production'
  target 'Project Production'

end
```

I don't have a config file for cocoapods and here the version i'm currently on : 
```
pod --version
1.0.0.beta.2
```

**Comment by mrackwitz** [comment]
Mh, this doesn't give me any hint, why the generated pod targets should be duplicated. Could you paste the outputs of `pod install --verbose`? This could give some further indication.

**Comment by lifely** [comment]
Hey @mrackwitz, i set the install to run and kinda forgot about it. 

Sorry about that, here you go: https://gist.github.com/lifely/7b6485b21eaf2e2d726c

**Comment by mrackwitz** [comment]
It seems like all targets are deduplicated as they should be. Don't worry about the aggregate targets. They don't increase the build time, as they only link together the targets in your Xcode project with the dependencies of all your pod dependencies. We have to use implicit dependencies with a target type which has an actual build product instead of aggregate targets as Xcode knows them, because we want to avoid having to add the `Pods` project as a subproject to the user's project as that cause among other things changing IDs on every installation.
If you should find pod targets to be actually duplicated, please send us more info, e.g. a full list of targets from Xcode. The cropping of the screenshot you've provided hides to much details.

**Comment by segiddins** [comment]
@mrackwitz is this an issue we can close?

**Comment by mrackwitz** [comment]
Yes, if it was really a problem, we could fix, there are good chances that #4146 has addressed it.

**Comment by maulikdhameliya** [comment]
Here is the solution https://stackoverflow.com/a/59007876/2053735
